src/plugins/location-finder/Box.vue

```diff
@@ -0,0 +1,42 @@
+<template>
+  <HorizontalCenter>
+    <Container>
+      <Spaced> </Spaced>
+      <Spaced>{{ location.id }} </Spaced>
+      <Spaced>
+        <Button
+          @click="ButtonClick()"
+          :disabled="location.indoors"
+        >
+        Download
+        </Button>
+      </Spaced>
+    </Container>
+  </HorizontalCenter>
+</template>
+
+<script>
+  export default {
+    props: {
+      location: {},
+      settings: {},
+    },
+    methods: {
+      ButtonClick() {
+        console.log( this.location )
+        if(this.location.id && (this.location.indoors == undefined)) {
+          let fileUrl = GameService.getLocationImgPath(this.location);
+          console.log(fileUrl);
+
+          let fileLink = document.createElement('a');
+          fileLink.href = fileUrl;
+          fileLink.setAttribute('download', this.location.id+ '.jpg');
+
+          document.body.appendChild(fileLink)
+          fileLink.click();
+          document.body.removeChild(fileLink);
```

Comment:
As a rule of thumb I'd like to avoid having plugins touch DOM outside of their own structure.

Can you please change that to perhaps having `<a href="..." ref="myLink">` that is in the plugin's template (with `display:none` style) that you then `this.$refs.myLink.click()`?

Reply:
Just made the changes. I believe that should do what you suggested. I didn't like the previous solution either, just didn't know the best way to do it (without new libraries).